### ISSUE TITLE
Plugin uses deprecated extension API version 1.0 (removed in GoCD 23.2.0)

### ISSUE THREAD

**Issue by chadlwilson** [szamfirov/gocd-okta-oauth-authorization-plugin#7]
Hiya @szamfirov  - GoCD maintainer courtesy call here. 👋

It seems this plugin is using the `1.0` extension API for authorizations.

https://github.com/szamfirov/gocd-okta-oauth-authorization-plugin/blob/c05e9a80027d5f481f34020a0c52a0ea1f07b3ea/src/main/java/cd/go/authorization/okta/Constants.java#L27-L28

Support for "1.0" will soon be removed in GoCD `23.2.0` as part of clean-up to support https://github.com/gocd/gocd/issues/11629 (functionality also likely of use to this plugin to increase security) Authorization extension API version `2.0` was introduced in GoCD `19.2.0` and was supposed to be removed **many** years ago, but was probably forgotten. 

https://plugin-api.gocd.org/current/#deprecations-across-extensions58

> Supports version 2 of [Authorization](https://plugin-api.gocd.org/current/authorization) extension API. Version 2 adds the following new messages:
> -   [Is Valid User](https://plugin-api.gocd.org/current/authorization#is-valid-user)
> -  [Get User Roles](https://plugin-api.gocd.org/current/authorization#get-user-roles)
>
> Plugins [Capabilities](https://plugin-api.gocd.org/current/authorization#get-plugin-capabilities) message is enhanced to add a new capability can_get_user_roles.
>
> The authorization extension v1 has been deprecated. This version will be removed in a release scheduled for September 2019. Plugin developers should upgrade their plugins to the latest version.

Although these additional messages are supported on V2, they are only used/required if you want GoCD access token-based authentication to work for users via the plugin, or if the the plugin declares capability `can_get_user_roles` e.g add on to the below

https://github.com/szamfirov/gocd-okta-oauth-authorization-plugin/blob/c05e9a80027d5f481f34020a0c52a0ea1f07b3ea/src/main/java/cd/go/authorization/okta/models/Capabilities.java#L24-L33

**Tl;DR** - I believe you can likely just declare "2.0" API version compatibility and things will work just fine as the server will assume the plugin does not support user roles, and does not support GoCD access/API token usage. This will work on GoCD versions `19.2.0+` so going back 3.5 years or so.

**Comment by szamfirov** [comment]
Hello @chadlwilson , thank you for raising this and also for providing all the extra information and references.

I've just released version [2.0.0](https://github.com/szamfirov/gocd-okta-oauth-authorization-plugin/releases/tag/2.0.0) for this plugin since it now points to the newer v2 authorization extension API. It's worth pointing out that it was done on a best effort basis - the builds and tests ran fine, however, I cannot test this in a running GoCD instance because I'm no longer actively using the product.

In terms of future maintenance, I'm more than happy to hand over the rains and transfer the ownership of this plugin to someone who will actively support this (perhaps in gocd-contrib) just to allow for better maintenance and avoid any delays in the future.

**Comment by tomcz** [comment]
I have been running the forked version of this plugin against GoCD 23.3.0 and can attest that it works as expected, or at least good enough for me not to see any difference in operation.

**Comment by chadlwilson** [comment]
Hi @szamfirov - I have invited you to the gocd-contrib org so you should have permissions to [transfer the repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository) into the org. Let me know if its not given as an option and I've missed a permission.

This way all of the releases, issues, PRs etc come with it. If you'd rather not do it this way, I can fork the repo there, but a repo transfer is usually the cleanest, as GitHub puts URL redirects into place and such :-)

We can keep it alive there from a basic perspective, alongside other plugins.

**Comment by szamfirov** [comment]
Thanks for arranging this on such short notice, @chadlwilson. I just made an attempt to transfer the repo but I've hit an insufficient permissions error:

<img width="844" alt="Screenshot 2023-08-10 at 16 55 37" src="https://github.com/szamfirov/gocd-okta-oauth-authorization-plugin/assets/4711683/fee958d7-93bd-4f03-b9a2-c932904e72e9">

**Comment by chadlwilson** [comment]
@szamfirov Sorry, fixed the permissions, can you please try again now?

**Comment by szamfirov** [comment]
There we go, transferred it 👍

**Comment by chadlwilson** [comment]
@szamfirov awesome! I had to make you an org member temporarily to allow the transfer but have now reinvited you as a collaborator on the repository.

When I get a bit of time I'll probably make some tweaks to automate the releasing and dependency upgrades consistent with other plugin repos and cut a new patch release to validate the process.

Thanks for your efforts with this plugin 🙏

**Comment by szamfirov** [comment]
> When I get a bit of time I'll probably make some tweaks to automate the releasing and dependency upgrades consistent with other plugin repos and cut a new patch release to validate the process.

Sounds like a good plan.

> Thanks for your efforts with this plugin 🙏

No worries at all, it was my pleasure. And who knows.. I might come back to this at some point as well :)

Btw, thank you for the nice gesture with updating the copyright 🤗
